### PR TITLE
Rework "Grid-based Navigation with Astar" demo

### PR DESCRIPTION
**Pull request by dalexeev**
* Use `AStarGrid2D` instead of `AStar3D`.
* Remove unused `tileset/tileset_source.tscn` file.

### CODE REVIEW

**Comment by Calinou** [comment]
Works great, but I can no longer teleport the player by holding <kbd>Shift</kbd> and clicking on an empty tile. Could this functionality be restored?

File: 2d/navigation_astar/project.godot [hunk 1]
```diff
@@ -24,14 +24,17 @@ window/stretch/aspect="expand"
...
-click={
+move_to={
 "deadzone": 0.5,
-"events": [Object(InputEventMouseButton,"resource_local_to_scene":false,"resource_name":"","device":0,"window_id":0,"alt_pressed":false,"shift_pressed":false,"ctrl_pressed":false,"meta_pressed":false,"button_mask":0,"position":Vector2(0, 0),"global_position":Vector2(0, 0),"factor":1.0,"button_index":1,"pressed":false,"double_click":false,"script":null)
+"events": [Object(InputEventMouseButton,"resource_local_to_scene":false,"resource_name":"","device":-1,"window_id":0,"alt_pressed":false,"shift_pressed":false,"ctrl_pressed":false,"meta_pressed":false,"button_mask":0,"position":Vector2(0, 0),"global_position":Vector2(0, 0),"factor":1.0,"button_index":1,"pressed":false,"double_click":false,"script":null)
+]
+}
+teleport_to={
+"deadzone": 0.5,
+"events": [Object(InputEventMouseButton,"resource_local_to_scene":false,"resource_name":"","device":-1,"window_id":0,"alt_pressed":false,"shift_pressed":true,"ctrl_pressed":false,"meta_pressed":false,"button_mask":1,"position":Vector2(250, 10),"global_position":Vector2(254, 53),"factor":1.0,"button_index":1,"pressed":true,"double_click":false,"script":null)
```
[the remark this answers is not in the document]
Comment (Calinou) [reply 2d/navigation_astar/project.godot:34]: cc @bruvzg

[the remark this answers is not in the document]
Comment (akien-mga) [reply 2d/navigation_astar/project.godot:34]: I'd suggest manually setting it to `Vector2(0, 0)` in this PR, and report the bug upstream. It probably comes from the new way to detect events in the input mapping dialog.

**Review by akien-mga — APPROVED** [review]
Should be good to merge, pending minor fix.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**